go-controller/pkg/ovn/base_network_controller_secondary.go

```diff
@@ -295,9 +249,29 @@ func (bsnc *BaseSecondaryNetworkController) addLogicalPortToNetworkForNAD(pod *k
 			pod.Namespace, pod.Name, nadName, time.Since(start), libovsdbExecuteTime)
 	}()
 
-	ops, lsp, podAnnotation, newlyCreated, err := bsnc.addLogicalPortToNetwork(pod, nadName, network)
-	if err != nil {
-		return err
+	// local pods need a logical port configured for them
+	// but remote pods on layer2 topologies with interconnect also need a remote
+	// logical port
+	isLocalPod := bsnc.isPodScheduledinLocalZone(pod)
+	requiresLogicalPort := isLocalPod || bsnc.isLayer2Interconnect()
```

Comment:
nit: maybe you could export this into a method / func ? If properly named you could drop the comment, plus I see this code is repeated further below.

Reply:
It's not repeated really. And I need both flags. So in this case I don't think having a method for this brings much benefit.